### PR TITLE
Update condition for max version limitation & add package tags

### PR DESCRIPTION
**Pull request by YuliiaKovalova**
## Fixes 
Updates restriction on the max version detection - previously it was capped by the environment version + add package metadata #101

### CODE REVIEW

File: src/MSBuildLocator/VisualStudioInstanceQueryOptions.cs [hunk 1]
```diff
@@ -33,5 +33,11 @@ public class VisualStudioInstanceQueryOptions
         ///     Working directory to use when querying for instances. Ensure it is the project directory to pick up the right global.json.
         /// </summary>
         public string WorkingDirectory { get; set; } = Environment.CurrentDirectory;
+
+        /// <summary>
+        ///     This variable enables the removal of the existing restriction on querying installed Visual Studio instances 
+        ///     with a runtime version lower than or equal to the one in the current environment.
+        /// </summary>
+        public bool AllowQueryAllRuntimeVersions { get; set; } = Environment.GetEnvironmentVariable("DOTNET_MSBUILD_QUERY_ALL_RUNTIMES") == "1";
```
Comment (rainersigwald) [inline src/MSBuildLocator/VisualStudioInstanceQueryOptions.cs:41]: What's motivating the environment variable here, instead of just requiring this setting?

[the remark this answers is not in the document]
Comment (rainersigwald) [reply src/MSBuildLocator/VisualStudioInstanceQueryOptions.cs:41]: @baronfel is there a good scenario for the inverse--.NET Framework locator finding SDK installations?

[the remark this answers is not in the document]
Comment (baronfel) [reply src/MSBuildLocator/VisualStudioInstanceQueryOptions.cs:41]: This would be part of the bucket of scenarios where the user is almost certainly not going to be able to _load_ the MSBuild, but they might just want to list/shell-invoke/etc the MSBuild? It could be useful, but a) wouldn't be as high of a priority in my mind, and b) would need to be behind whatever 'I promise I know this won't work if I try to in-proc this' flag/gate we've set up here.

Comment (YuliiaKovalova) [reply src/MSBuildLocator/VisualStudioInstanceQueryOptions.cs:41]: I wanted to avoid extending public API. 
If adding a flag is a better option I can change.

Comment (YuliiaKovalova) [reply src/MSBuildLocator/VisualStudioInstanceQueryOptions.cs]: fixed

File: src/MSBuildLocator/MSBuildLocator.cs [hunk 2]
```diff
@@ -43,6 +43,12 @@ public static class MSBuildLocator
         /// </summary>
         public static bool IsRegistered => s_registeredHandler != null;
 
+        /// <summary>
+        ///     This flag enables the removal of the existing restriction on querying installed Visual Studio instances 
+        ///     with a runtime version lower than or equal to the one in the current environment.
+        /// </summary>
```
Comment (rainersigwald) [inline src/MSBuildLocator/MSBuildLocator.cs:49]: I would like to rephrase this to be scarier :)

```suggestion
        /// <summary>
        ///     Allow discovery of .NET SDK versions that are unlikely to be successfully loaded in the current process.
        /// </summary>
        /// <remarks>
        ///     Defaults to <see langword="false"/>. Set this to <see langword="true"/> only if your application has special logic to handle loading an incompatible SDK, such as launching a new process with the target SDK's runtime.
        /// </remarks.
```

File: src/MSBuildLocator/DotNetSdkLocationHelper.cs [hunk 3]
```diff
@@ -56,8 +56,8 @@ internal static class DotNetSdkLocationHelper
             // in the .NET 5 SDK rely on the .NET 5.0 runtime. Assuming the runtime that shipped with a particular SDK has the same version,
             // this ensures that we don't choose an SDK that doesn't work with the runtime of the chosen application. This is not guaranteed
             // to always work but should work for now.
-            if (major > Environment.Version.Major ||
-                (major == Environment.Version.Major && minor > Environment.Version.Minor))
+            if (major > Environment.Version.Major || (major == Environment.Version.Major && minor > Environment.Version.Minor) 
+                && !allowQueryAllRuntimeVersions)
```
Comment (rainersigwald) [inline src/MSBuildLocator/DotNetSdkLocationHelper.cs:60]: I can never remember boolean operator precedence rules. How would you feel about reordering this?

```suggestion
            if (!allowQueryAllRuntimeVersions && 
                (major > Environment.Version.Major || 
                 (major == Environment.Version.Major && minor > Environment.Version.Minor)))
```

**Review by rainersigwald — APPROVED** [review]
A couple of nits but looking good IMO. Can you give a minor-version bump in `version.json` since this is a feature addition?

File: src/MSBuildLocator/MSBuildLocator.cs [hunk 4]
```diff
@@ -43,6 +43,14 @@ public static class MSBuildLocator
         /// </summary>
         public static bool IsRegistered => s_registeredHandler != null;
 
+        /// <summary>
+        ///     Allow discovery of .NET SDK versions that are unlikely to be successfully loaded in the current process.
+        /// </summary>
+        /// <remarks>
+        ///     Defaults to <see langword="false"/>. Set this to <see langword="true"/> only if your application has special logic to handle loading an incompatible SDK, such as launching a new process with the target SDK's runtime.
+        /// </remarks.
+        public static bool AllowQueryAllRuntimeVersions { get; set; } = false;
```
Comment (jasonmalinowski) [inline src/MSBuildLocator/MSBuildLocator.cs:52]: Sorry I only noticed this PR now, but @YuliiaKovalova should a flag like this have been added to the VisualStudioInstanceQueryOptions rather than being a global static? Otherwise it's a bit unclear what a library like Roslyn's MSBuildWorkspace should do here since technically we're just a library and won't ever be registering MSBuild in the calling applications' process at this point -- do we set this and set it back to the old value, and pray there wasn't another thread doing something?

Comment (rainersigwald) [reply src/MSBuildLocator/MSBuildLocator.cs:52]: Oh, I completely missed that move! I agree with @jasonmalinowski that this made more sense on `VisualStudioInstanceQueryOptions`.
